### ISSUE TITLE
Mapping of Values for Raw Test Result and Impact

### ISSUE THREAD

**Issue by LBKo** [Vulnerator/Vulnerator#82]
I'm using Vulnerator 6.1.7. I imported a STIG result (Win10 ckl file) and a nessus scan of the target and generated a Vulnerator report. For the STIG result, I looked at the RAR worksheet and the STIG Details worksheet. The RAR's Raw Test Result column (values I, II, III) appears to equate to the STIG Details' STIG Severity column, while the RAR's Impact column (values L, M, H) appears to equate to the STIG Details' Risk Factor column. For the ACAS result, the same mappings of columns appear to apply for the RAR worksheet and the ACAS Output worksheet.
1. The DISA STIG has a Severity field, but no Risk Factor field. Does Vulnerator populate Risk Factor based on Severity I equal to Risk Factor H, a II to a M, and a III to a L? (I believe so, but want to confirm.)
2. The ACAS Output worksheet does not appear to have a similar relationship between Risk Factor and Severity. What is the connection between ACAS Output Risk Factor and Severity columns? (The Vulnerator guide indicates that Tenable could provide STIG Severity values.)

**Comment by amkuchta** [comment]
@LBKo I don't think that I am following - can you please email a _**sanitized**_ version (no IP addresses or host names!) of the report to alex.kuchta@navy.mil? I'd like to get an idea of what exactly you are looking at.

**Comment by LBKo** [comment]
Hi Alex

Here's the Vulnerator 6.1.7 report. I used an ACAS nessus file and a couple of ckl files from STIG Viewer. Within Vulnerator, I chose RMF, 800-53 Rev 4, and then I think I kept all other default settings. The nessus files are approximately 5 months old, and I did not run those scans. Same for the ckl files.

When comparing the report's RAR and STIG Details worksheets, I think the RAR's Raw Test Result column = STIG Details' STIG Severity column. Both columns have values I, II, III, and those values are from the STIG's Severity values (e.g., CAT I, CAT II, CAT III). The RAR's Impact column seems to be the same as the STIG Details' Risk Factor column. Both have High, Medium, Low. Not sure of the source for the Risk Factor column. Is Risk Factor assigned based on I = High, II = Medium, III = Low?

When comparing the report's RAR and ACAS Output & Review worksheets, I think the RAR's Raw Test Result column = ACAS Output & Review's STIG Severity column. Both columns have values I, II, III. The RAR's Impact column seems to the be same as the ACAS Output & Review's Risk Factor column, but not all of the STIG Severity values are filled in on the ACAS Output & Review rows. For example, plugin 11457 on the RAR tab has Raw Test Result = IV and Impact L; but the ACAS Output & Review worksheet shows the four entries for 11457 as each having Risk Factor = Low but not STIG Severity value. Also, in the ACAS Output & Review worksheet, I see Risk Factor = Medium having STIG Severity = I and Low = <blank>. So I am trying to determine the connection between the ACAS results for these columns in the two worksheets.

Appreciate your help.

v/r

Leo B. Kotowski, CISSP
SimVentions, Inc.
100 Riverside Parkway, Suite 123
Fredericksburg, VA 22406
(540) 370-8341


-----Original Message-----

**Comment by LBKo** [comment]
Leo Kotowski would like to recall the message, "[Vulnerator/Vulnerator] Mapping of Values for Raw Test Result and Impact (#82)".

**Comment by amkuchta** [comment]
@LBKo, I got the file - thank you!  To address your issues:

- You are correct - DISA STIG files have a one to one mapping:
  - High = CAT I
  - Medium = CAT II
  - Low = CAT III
- Things get a bit more difficult with ACAS - some findings have a "DISA STIG Severity" (CAT I/II/III) assigned by Tenable.  If this value exists, it is used to identify the Raw Risk and Impact of the plugin using the exact same mapping as above.  However, if no DISA STIG Severity is found, the Vulnerator defaults back to using the ACAS-provided risk factor information with the following mapping:
  - Critical / High = CAT I
  - Medium = CAT II
  - Low = CAT III
  - Informational = CAT IV

For that reason, not all items on the "ACAS Output & Review" report will have a "DISA STIG Severity" assigned. 

Please let me know if this helps!

**Comment by LBKo** [comment]
Alex

Thanks much for explaining. Most definitely helps. I, and other co-workers who use Vulnerator extensively, were not sure how Vulnerator correlated the ACAS results. I'll relay this info to them.

v/r

Leo B. Kotowski, CISSP
SimVentions, Inc.
100 Riverside Parkway, Suite 123
Fredericksburg, VA 22406
(540) 370-8341


-----Original Message-----

**Comment by LBKo** [comment]
Alex

Another related question... up front. How is Vulnerator's Impact field value determined by digesting ckl and nessus files?

I have used Vulnerator 6.1.7 with ckl and nessus files. On the Vulnerator RAR tab, I noticed the Impact column was prepopulated with risk factor values -- VL, L, M, H, VH.

Per the Navy's RAR template, Impact is "Determine the adverse impact (i.e., potential harm to organizational operations, organizational assets, individuals, other organizations, or the Nation) from the threat event. Risk assessors may consider the Concept of Operations, system description, User Representative input, etc. (See Table 6 on the RMF KS page "Model for Assessing Residual Risk Level for Non-Compliant Security Controls" for descriptions of the values assigned to impact.)"

The Navy's RAR template seems to indicate Impact would be a selectable value by the risk assessor.

v/r

Leo B. Kotowski, CISSP
SimVentions, Inc.
100 Riverside Parkway, Suite 123
Fredericksburg, VA 22406
(540) 370-8341


-----Original Message-----
From: Leo Kotowski 
Sent: Thursday, January 26, 2017 1:57 PM
To: 'Vulnerator/Vulnerator' <reply@reply.github.com>
Cc: 'alex.kuchta@navy.mil' <alex.kuchta@navy.mil>
Subject: RE: [Vulnerator/Vulnerator] Mapping of Values for Raw Test Result and Impact (#82)

Alex

Thanks much for explaining. Most definitely helps. I, and other co-workers who use Vulnerator extensively, were not sure how Vulnerator correlated the ACAS results. I'll relay this info to them.

v/r

Leo B. Kotowski, CISSP
SimVentions, Inc.
100 Riverside Parkway, Suite 123
Fredericksburg, VA 22406
(540) 370-8341


-----Original Message-----
